### DISCUSSION TITLE
Wrapper around klog for billing (FYI)

### DISCUSSION THREAD

**Discussion by MoonCactus** [jotaen/klog#265] [Show and tell]
Just FYI I just published the script that makes me even more productive with `klog`.

It started with an alias, ended up as a growing (hence dirty) bash script.

  * one centralized journal for all my clients
  * **billing** with decimal hours 
  * which implied filtering clients and tasks

Sadly, I need to admit that `tt` has now diverged too much from the evolving `klog`.
Nastier even, my script would break on many of less-often-used `klog` features.

Anyhow it might be of interest to see how it works in order to improve klog if it makes sense

 https://github.com/MoonCactus/tt

Eg:
```
  tt client1=sysadmin "upgraded dokuwiki"  # start doing some sysadmin for client 1
  tt client1=dev "bug fix 123" # stop previous task and start another one
  tt client2=phone             # stop previous task and start a new one (other client)
  tt end                       # stop timetracking (eg. for lunch, or on shutdown - klog break or negative times are NOT supported)
  tt resume client1.*wiki      # resume former matching task (we had a discussion on the topic, but tt moves a bit further than recent klog "resume" implementation)

  tt bill client1              # cumulative time spent en client1 (new and just so useful when I bill my clients !)
  tt bill client1=sysadmin     # time spent on client1 specific task

  tt backup
  tt keep 'client1=(dev.*|sysadmin)'| tt bill client1 > billing_client1_dev_and_sysadmin_only.txt
  tt discard 'client1=(dev.*|sysadmin)'| tt replace  # remove previous billed entries from journal
```

I still think that command-line time tracking beats everything I ever tried, and I still think `klog` shines in this respect. Kudos.

**Comment by jotaen** [comment]
Awesome, thanks for sharing this! It’s cool to see how you wrapped the klog binary in order to customise it for your personal needs and workflows. I think this can be a good inspiration for others who consider to do something similar.
